elytron-oidc-client/src/test/java/org/wildfly/extension/elytron/oidc/ElytronOidcClientSubsystemTransformerTestCase.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.extension.elytron.oidc;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.model.test.FailedOperationTransformationConfig;
+import org.jboss.as.model.test.ModelTestControllerVersion;
+import org.jboss.as.model.test.ModelTestUtils;
+import org.jboss.as.subsystem.test.AbstractSubsystemSchemaTest;
+import org.jboss.as.subsystem.test.AdditionalInitialization;
+import org.jboss.as.subsystem.test.KernelServices;
+import org.jboss.as.subsystem.test.KernelServicesBuilder;
+import org.jboss.as.version.Stability;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.wildfly.extension.elytron.oidc.OidcTestCase.DefaultInitializer;
+
+@RunWith(value = Parameterized.class)
+public class ElytronOidcClientSubsystemTransformerTestCase extends AbstractSubsystemSchemaTest<ElytronOidcSubsystemSchema> {
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> parameters() {
+        return List.<Object[]>of(
+                new Object[] { ModelTestControllerVersion.EAP_8_0_0, ElytronOidcClientSubsystemModel.VERSION_2_0_0.getVersion() }
+        );
+    }
+    private final ModelTestControllerVersion controllerVersion;
+    private final ModelVersion modelVersion;
+    private static final PathAddress SUBSYSTEM_ADDRESS = PathAddress.pathAddress(ModelDescriptionConstants.SUBSYSTEM, ElytronOidcExtension.SUBSYSTEM_NAME);
+
+
+    public ElytronOidcClientSubsystemTransformerTestCase(ModelTestControllerVersion controllerVersion, ModelVersion version) {
+        super(ElytronOidcExtension.SUBSYSTEM_NAME, new ElytronOidcExtension(), ElytronOidcSubsystemSchema.VERSION_2_0_PREVIEW, ElytronOidcSubsystemSchema.CURRENT.get(Stability.PREVIEW));
+        this.controllerVersion = controllerVersion;
+        this.modelVersion = version;
+    }
+
+    @Test
+    public void testTransformations() throws Exception {
+        final ModelVersion version = modelVersion;
```

Comment:
Other transformer tests I saw use `controller.getSubsystemModelVersion(getMainSubsystemName())`, but `ModelTestControllerVersion.EAP_8_0_0` does not have a field for elytron-oidc-client, and the above returns null, so I had to use `modelversion`.